### PR TITLE
Ts api improvements

### PR DESCRIPTION
**Pull request by davebx**
- Paginate repositories in category if count exceeds set number
 - Allow sorting by name, owner, description
 - Add a numeric_revision column to the changeset metadata table

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/toolshed.py [hunk 1]
```diff
@@ -188,9 +188,12 @@ def category(self, trans, **kwd):
         :param tool_shed_url: the url of the toolshed to get repositories from
         :param category_id: the category to get repositories from
         """
+        sort_order = kwd.get('sort_order', 'asc')
+        sort_key = kwd.get('sort_key', 'name')
+        page = kwd.get('page', 1)
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/api/toolshed.py:196]: Can you add the new parameters to the docstring?

**Comment by dannon** [comment]
@davebx Are the toolshed test failures related, here?

**Comment by davebx** [comment]
@dannon I doubt it, but I haven't actually verified that.

**Comment by nsoranzo** [comment]
@davebx I've restarted the toolshed tests, and exactly the same 3 tests failed. I've rarely seen TS tests fail randomly lately, so I fear these are related to the PR. Maybe you can try to rebase the PR and `git bisect` to find which commit broke the tests?

**Comment by jmchilton** [comment]
Marking as WIP due to the test failures. I'm kicking it for now, but happy to review and try to get it into 19.01 if the tests can be fixed sooner.

**Comment by davebx** [comment]
Well, at least the _toolshed_ tests passed this time.

**Pull request merged**